# Module 1 Lab 2 — Hands-On Problem 1.1

## Which feed is best for business?

This lab compares minimum, mean, and maximum chick weights for each feed type. The business assumption is that heavier chickens are better.

In [1]:
import pandas as pd

df = pd.read_csv('chickwts.csv')
print(f'Loaded {len(df)} rows from chickwts.csv')

Loaded 71 rows from chickwts.csv\n

In [2]:
df.head()

 Unnamed: 0  weight      feed
          1     179 horsebean
          2     160 horsebean
          3     136 horsebean
          4     227 horsebean
          5     217 horsebean
          6     168 horsebean
          7     108 horsebean
          8     124 horsebean
          9     143 horsebean
         10     140 horsebean
         11     309   linseed
         12     229   linseed
         13     181   linseed
         14     141   linseed
         15     260   linseed
         16     203   linseed
         17     148   linseed
         18     169   linseed
         19     213   linseed
         20     257   linseed
         21     244   linseed
         22     271   linseed
         23     243   soybean
         24     230   soybean
         25     248   soybean
         26     327   soybean
         27     329   soybean
         28     250   soybean
         29     193   soybean
         30     271   soybean
         31     316   soybean
         32     267   soybean
         3

In [3]:
# Calculate minimum, mean, and maximum weight for each feed type.
summary = (
    df.groupby('feed')['weight']
      .agg(['min', 'mean', 'max'])
      .reset_index()
      .sort_values('mean', ascending=False)
)
summary['mean'] = summary['mean'].round(2)
summary

     feed  min   mean  max
sunflower  226 328.92  423
   casein  216 323.58  404
 meatmeal  153 276.91  380
  soybean  158 246.43  329
  linseed  141 218.75  309
horsebean  108 160.20  227


## Recommendation

Based on the summary statistics, I would choose **sunflower feed** for the business. It has the highest average chick weight at **328.92**, so it produces the heaviest chickens on average. It also has the highest minimum weight at **226** and the highest maximum weight at **423**, which makes its results strong at both extremes. Casein is the closest alternative because its mean weight is **323.58**, but its minimum and maximum weights are both lower than sunflower feed. Because sunflower performs best on the average while also having the strongest minimum and maximum, I would buy sunflower feed.